addons/website_sale_stock/models/sale_order.py

```diff
@@ -105,7 +105,7 @@ def _are_all_product_available_for_purchase(sale_order):
                 cache_key = self._get_cache_key_for_line(line)
```

Comment:
I checked this with a test and this issue does not occur because of the fact that one sale_order (abandonned cart) is checked at a time.

I added this in the test
```
        company_2 = self.env['res.company'].create({'name': 'Company D'})
        website_2 = self.env['website'].create({'name': 'Website Company D'})
        website_2.send_abandoned_cart_email = True
        website_2.company_id = company_2
        warehouse_2 = self.env['stock.warehouse'].search([('company_id', '=', company_2.id)])
        customer_d = self.env['res.partner'].create({
            'name': 'd',
            'email': 'd@example.com',
        })
        self.env['stock.quant'].with_context(inventory_mode=True).create([{
            'product_id': product.id,
            'inventory_quantity': 25.0,
            'location_id': warehouse_2.lot_stock_id.id,
        }]).action_apply_inventory()
        sale_order_2 = self.env['sale.order'].create({
            'partner_id': customer_d.id,
            'website_id': website_2.id,
            'state': 'draft',
            'date_order': (datetime.utcnow() - relativedelta(hours=website.cart_abandoned_delay)) - relativedelta(
                minutes=1),
            'order_line': order_line
        })

        self.assertTrue(self.send_mail_patched(sale_order_2.id))
```

mail about sale_order_2 will not send only if a change  'inventory_quantity' to 0.0, which is correct.